### ISSUE TITLE
`check_auth` permissions discrepancy

### ISSUE THREAD

**Issue by tjlevel12** [level12/keg-auth#59]
Currently, if a Keg view implements a `check_auth` method, that method can be used to determine if the user should have access to the view (ie, should a 403 be returned).

Unfortunately, the nav system doesn't seem to have any way of accessing the `check_auth` instance method. As a result, the nav system can include links that return a 403 when clicked.

I don't currently see any way to resolve this with the `check_auth` method; permissions must instead be specified using the `require_permissions` decorator.

If using `check_auth` can't be resolved, we should add an "Upgrading to keg-auth" section to the readme which specifies that `check_auth` should not be used.

**Comment by guruofgentoo** [comment]
TLDR: the section of the readme that references nav helpers should note that `check_auth` logic will not be referenced when generating the nav tree, since it is run on the view instance.

We don't want nav generation to instantiate all referenced view classes to get that info, and doing so may not have enough information for the logic anyway. Having the link in nav is not intended to be a guarantee that the target is accessible - just that with auth and nav using all the tools in keg-auth, most cases will line up that way.

**Comment by guruofgentoo** [comment]
This is addressed in documentation for #59.
